etc/xcc-router/src/lib.rs

```diff
@@ -225,4 +229,120 @@ impl Router {
+                        id
+                    }
+                }
+            }
+            NearPromise::And(promises) => {
+                let indices: Vec<PromiseIndex> = promises
+                    .iter()
+                    .map(Self::recursive_promise_create)
+                    .collect();
+                env::promise_and(&indices)
+            }
+        }
+    }
+
+    fn add_batch_actions(id: PromiseIndex, actions: &[PromiseAction]) {
+        for action in actions.iter() {
+            match action {
+                PromiseAction::CreateAccount => env::promise_batch_action_create_account(id),
+                PromiseAction::Transfer { amount } => {
+                    env::promise_batch_action_transfer(id, amount.as_u128())
+                }
+                PromiseAction::DeployConotract { code } => {
+                    env::promise_batch_action_deploy_contract(id, code)
+                }
+                PromiseAction::FunctionCall {
+                    name,
+                    args,
+                    attached_yocto,
+                    gas,
+                } => env::promise_batch_action_function_call(
+                    id,
+                    name,
+                    args,
+                    attached_yocto.as_u128(),
+                    gas.as_u64().into(),
+                ),
+                PromiseAction::Stake { amount, public_key } => {
+                    env::promise_batch_action_stake(id, amount.as_u128(), &to_sdk_pk(public_key))
+                }
+                PromiseAction::AddFullAccessKey { public_key, nonce } => {
+                    env::promise_batch_action_add_key_with_full_access(
+                        id,
+                        &to_sdk_pk(public_key),
+                        *nonce,
+                    )
+                }
+                PromiseAction::AddFunctionCallKey {
+                    public_key,
+                    nonce,
+                    allowance,
+                    receiver_id,
+                    function_names,
+                } => {
+                    let receiver_id = near_sdk::AccountId::new_unchecked(receiver_id.to_string());
+                    env::promise_batch_action_add_key_with_function_call(
+                        id,
+                        &to_sdk_pk(public_key),
+                        *nonce,
+                        allowance.as_u128(),
+                        &receiver_id,
+                        function_names,
+                    )
+                }
+                PromiseAction::DeleteKey { public_key } => {
+                    env::promise_batch_action_delete_key(id, &to_sdk_pk(public_key))
+                }
+                PromiseAction::DeleteAccount { beneficiary_id } => {
+                    let beneficiary_id =
+                        near_sdk::AccountId::new_unchecked(beneficiary_id.to_string());
+                    env::promise_batch_action_delete_account(id, &beneficiary_id)
+                }
```

Comment:
Yes, this does give developers the ability to brick their own router account. But without these functions, they also cannot create subaccounts of their router account. I don't know that there is any real use-case for that or not, but I was aiming for completeness in terms of being able to do anything with a router account that you could do with a normal NEAR account.

Maybe this is worth discussing on an engineering call sometime? If people think this is too dangerous then I can remove it.

Reply:
I've hidden this functionality behind a feature flag: https://github.com/aurora-is-near/aurora-engine/pull/610/commits/b22a05cb1de50c187543d2ccc7db313395ecefd2

So it will not be enabled for now, but if someone does present a legitimate use-case for it then we can deliver the functionality quickly by simply enabling the feature flag.